l3kernel/l3token.dtx

```diff
@@ -2245,6 +2282,46 @@
 %    \end{macrocode}
 % \end{macro}
 %
+% \begin{macro}[pTF]{\token_if_control_word:N}
+% \begin{macro}{\@@_if_control_word_aux:w}
+%   Check if token consists of the escape character followed by
+%   one or more letters.
+%   Hence we do not check catcode, we have to check if it is a
+%   character whose catcode is parameter, or much easier, we just test
+%   if it is a character. And |#1| could be a control symbol let to
+%   \cs{else:} or \cs{fi:}, etc., so we use a auxiliary function
+%   to grab second \cs{if:w}.
+%    \begin{macrocode}
+\prg_new_conditional:Npnn \token_if_control_word:N #1 { p , T , F , TF }
+  {
+    \if:w 0 \tex_strcmp:D { \exp_not:N #1 } { \token_to_str:N #1 }
+      \@@_if_control_word_aux:w
+    \else:
+      \if:w \exp_not:N #1 \token_to_str:N #1
+        \prg_return_false: \else: \prg_return_true: \fi:
+    \fi:
+  }
+\cs_new:Npn \@@_if_control_word_aux:w #1 \fi: { \prg_return_false: }
+%    \end{macrocode}
+% \end{macro}
+% \end{macro}
+% \begin{macro}[pTF]{\token_if_control_symbol:N}
+%   A similar plan but we need to test for the space behavior the whether we
```

Comment:
The "the" right before "whether" doesn't read well. Perhaps it should be "of"?

Reply:
"of" seems unlikely.
My guess (probably bad) is "

Reply:
But now I see that it should be:
  and then
Or maybe 
, and then

OK?